frontend-react/src/components/StaticCompare.tsx

```diff
@@ -0,0 +1,143 @@
+import { ReactElement, useCallback, useEffect, useState } from "react";
+import DOMPurify from "dompurify";
+import { ScrollSync, ScrollSyncPane } from "react-scroll-sync";
+
+import { Diff, SES_TYPE } from "../utils/diff";
+import { splitOn } from "../utils/misc";
+
+interface StaticCompareProps {
+    rightText: string;
+    leftText: string;
+}
+
+const insertHighlight = (
+    s1: string,
+    offset: number,
+    length: number
+): string => {
+    if (s1 === "" || length === 0) {
+        return "";
+    }
+    // we want to insert a <span></span> around text.
+    const three_parts = splitOn(s1, offset, offset + length);
```

Comment:
[nit] `threeParts` - imo snake case has no place in js variable naming outside of all caps constants